src/assets/Azure.Core.Shared/PageResponseEnumerator.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal static class PageResponseEnumerator
+    {
+        public static FuncPageable<T> CreateEnumerable<T>(Func<string?, Page<T>> pageFunc) where T : notnull
+        {
+            return new FuncPageable<T>((continuationToken, pageSizeHint) => pageFunc(continuationToken));
+        }
+
+        public static FuncPageable<T> CreateEnumerable<T>(Func<string?, int?, Page<T>> pageFunc) where T : notnull
```

Comment:
Do we need all the overloads?

Reply:
The overloads are to be consistent with the original public interfaces. Since it is Azure.Core, I figured others may need want the additional flexibility. Also, I didn't know what interfaces the generator might need. Right now, I'm not using all of them. If these were only intended for the generator, I'd likely design the interface a bit differently. For example, the delegates I created would be part of the public interface instead of using different Funcs for everything.

Reply:
we have a type that's used by non generated code, this is purely for generator.